pkg/deploy/controller/lifecycle/manager.go

```diff
@@ -0,0 +1,164 @@
+package lifecycle
+
+import (
+	"fmt"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+)
+
+// Status represents the status of a lifecycle action, and should be treated
+// as the source of truth by controllers.
+type Status string
+
+const (
+	// Pending means the action has not yet executed.
+	Pending Status = "Pending"
```

Comment:
So, these annotations are kind of an implementation detail of the lifecycle plugin (execnewpod), but I can see how users might want to look at them. I can make them public.

Similarly: the plugin has its own custom status types (persisted), which the plugin knows how to adapt back into the generic lifecycle status values (not persisted). The latter are what the other controllers know about directly. If useful, I could also store the generic lifecycle status values with the deployment for UIs to present, but I can't decide if that's redundant when the UI will have access to the plugin-specific annotations.